afs/afs-ws/afs-ws-client/src/main/java/com/powsybl/afs/ws/client/RemoteAppFileSystemProvider.java

```diff
@@ -24,21 +29,30 @@
 @AutoService(AppFileSystemProvider.class)
 public class RemoteAppFileSystemProvider implements AppFileSystemProvider {
 
-    private final List<RemoteAppFileSystemConfig> configs;
+    private final Supplier<RemoteServiceConfig> config;
 
     public RemoteAppFileSystemProvider() {
-        this.configs = RemoteAppFileSystemConfig.load();
+        this(RemoteServiceConfig.INSTANCE);
+    }
+
+    public RemoteAppFileSystemProvider(Supplier<RemoteServiceConfig> config) {
+        this.config = Objects.requireNonNull(config);
     }
 
     @Override
-    public List<AppFileSystem> getFileSystems(ComputationManager computationManager) {
-        return configs.stream()
-                .map(config ->  {
-                    RemoteAppStorage storage = new RemoteAppStorage(config.getFileSystemName(), config.getUrl());
-                    RemoteListenableAppStorage listenableStorage = new RemoteListenableAppStorage(storage, config.getUrl());
-                    RemoteTaskMonitor taskMonitor = new RemoteTaskMonitor(config.getFileSystemName(), config.getUrl());
-                    return new AppFileSystem(config.getFileSystemName(), true, listenableStorage, taskMonitor);
-                })
-                .collect(Collectors.toList());
+    public List<AppFileSystem> getFileSystems(AppFileSystemProviderContext context) {
+        if (context.getToken() == null) {
+            return Collections.emptyList();
+        } else {
+            URI uri = config.get().getRestUri();
```

Comment:
config.get() could return a null instance: you should check it

Reply:
Done